internal/ssm/install_test.go

```diff
@@ -0,0 +1,132 @@
+package ssm_test
+
+import (
+	"context"
+	"net/http"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/ProtonMail/gopenpgp/v3/crypto"
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/zap"
+
+	"github.com/aws/eks-hybrid/internal/ssm"
+	"github.com/aws/eks-hybrid/internal/test"
+	"github.com/aws/eks-hybrid/internal/tracker"
+)
+
+func generateKeyPair(t *testing.T) (string, *crypto.Key) {
+	pgp := crypto.PGP()
+	key, err := pgp.KeyGeneration().
+		AddUserId("test", "test@example.com").
+		New().GenerateKey()
+	assert.NoError(t, err)
+
+	armoredPublicKey, err := key.GetArmoredPublicKey()
+	assert.NoError(t, err)
+
+	return armoredPublicKey, key
+}
+
+func generateSignature(t *testing.T, key *crypto.Key, data []byte) []byte {
+	pgp := crypto.PGP()
+	signer, err := pgp.Sign().SigningKey(key).Detached().New()
+	assert.NoError(t, err)
+	signature, err := signer.Sign(data, crypto.Bytes)
+	assert.NoError(t, err)
+	return signature
+}
+
+func TestInstall(t *testing.T) {
+	publicKey, privateKey := generateKeyPair(t)
+	_, wrongPrivateKey := generateKeyPair(t)
+
+	installerData := []byte("#!/bin/bash\necho 'test installer'")
+
+	tests := []struct {
+		name          string
+		installerData []byte
+		signature     []byte
+		statusCode    int
+		wantErr       bool
```

Comment:
a more robust way to do this is to have here the error message as a string (or part of the message)

That way not only you test that you get an error when you should, you check that you get the error that you expect getting